app/services/CharacterService.js

```diff
@@ -0,0 +1,51 @@
+const rp = require("request-promise");
+const slug = require("slug");
+const {
+    NAMESPACE_PROFILE_US,
+    DEFAULT_LOCALE
+} = require("../constants");
+
+class CharacterService {
+
+    constructor(oauthClient) {
+        this.oauthClient = oauthClient;
+    }
+
+    async getCharacter(characterName, realmName) {
+        const oauthToken = await this.oauthClient.getToken();
+        const encodedCharacterName = encodeURIComponent(characterName);
+        const realmNameSlug = slug(realmName);
+        const characterSummaryDocumentURL = `https://us.api.blizzard.com/profile/wow/character/${realmNameSlug}/${encodedCharacterName}`;
+        const response = await rp.get({
```

Comment:
We've since moved to `node-fetch` which covers the error handling case a bit more.

```
        const response = await fetch(documentUri, { headers });
        if (!response.ok) {
            throw new Error(`${response.status}: ${response.statusText}: ${documentUri}`);
        }
        return response.json();
```